lib/logstash/pluginmanager/util.rb

```diff
@@ -1,64 +1,101 @@
+module LogStash::PluginManager
 
-class LogStash::PluginManager::Util
-
-  def self.logstash_plugin?(gem)
+  # check for valid logstash plugin gem name & version or .gem file, logs errors to $stdout
+  # uses Rubygems API and will remotely validated agains the current Gem.sources
+  # @param plugin [String] plugin name or .gem file path
+  # @param version [String] gem version requirement string
+  # @return [Boolean] true if valid logstash plugin gem name & version or a .gem file
+  def self.is_logstash_plugin?(plugin, version = nil)
+    if is_plugin_file?(plugin)
+      begin
+        return is_logstash_plugin_gem_spec?(plugin_file_spec(plugin))
+      rescue => e
+        $stderr.puts("Error reading plugin file #{plugin}, caused by #{e.class}")
+        $stderr.puts(e.message) if ENV["DEBUG"]
+        return false
+      end
+    else
+      dep = Gem::Dependency.new(plugin, version || Gem::Requirement.default)
+      specs, error = Gem::SpecFetcher.fetcher.spec_for_dependency(dep)
 
-    gem_data = case
-    when gem.is_a?(Gem::Specification); gem
-    when (gem =~ /\.gem$/ and File.file?(gem)); Gem::Package.new(gem).spec
-    else Gem::Specification.find_by_name(gem)
+      # depending on version requirements, multiple specs can be returned in which case
+      # we will grab the one with the highest version number
+      if latest = specs.map{|tuple| tuple.first}.max_by{|spec| spec.version}
```

Comment:
let's leave assignment for now...... gruuuumble :P

We'll duke it out when we figure out our rubocop rules. Deal?